### PR TITLE
Remove Controller.ViewEngine property+unit tests and simplify Content method

### PR DESCRIPTION
**Pull request by ianhong**
@yishaigalatzer and @rynowak 

As we discussed I removed ViewEngine from Controller. Plus, I updated Content method too. 
All tests are passed after removing the property. 

Thanks,

### CODE REVIEW

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Core/Controller.cs [hunk 1]
```diff
@@ -284,18 +263,10 @@ public virtual ContentResult Content(string content, string contentType, Encodin
...
-            if (contentType != null)
-            {
-                result.ContentType = contentType;
-            }
-
-            if (contentEncoding != null)
-            {
-                result.ContentEncoding = contentEncoding;
-            }
-
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/Controller.cs]: what's the purpose of this change? just simplifying the code?

File: test/Microsoft.AspNet.Mvc.Core.Test/ControllerTests.cs [hunk 2]
```diff
@@ -1238,65 +1238,6 @@ public void ControllerDispose_CallsDispose()
...
-
-            // Act
-            var unsused = controller.ViewEngine;
-            var result = controller.View();
-
-            // Assert
-            Assert.Same(viewEngine, result.ViewEngine);
-        }
-
         private static Controller GetController(IModelBinder binder, IValueProvider provider)
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/ControllerTests.cs]: Would like to see a unit test added for the `View(...)` method on controller - since that's the purpose of this change :grinning:

**Comment by rynowak** [comment]
:watch:

File: src/Microsoft.AspNet.Mvc.Core/Controller.cs (hunk 1 above)
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Core/Controller.cs]: just simplification, ContentResult just have auto properties for these values so null checking is meaningless

File: test/Microsoft.AspNet.Mvc.Core.Test/ControllerTests.cs (hunk 2 above)
Comment (yishaigalatzer) [inline test/Microsoft.AspNet.Mvc.Core.Test/ControllerTests.cs]: :+1:

**Comment by rynowak** [comment]
:shipit: from me, please get sign off from @yishaigalatzer as well.

**Comment by ianhong** [comment]
@yishaigalatzer as Ryan said I have the unit test for View in the other PR, #1820. Please sign off this one if possible.

**Comment by yishaigalatzer** [comment]
:shipit:
